gibbon-compiler/src/Gibbon/L2/Syntax.hs

```diff
@@ -74,6 +74,29 @@ type Ty2 = UrTy LocVar
 -- | Shorthand for recursions.
 type E2 l d = PreExp E2Ext l d
 
+data RegionSize = BoundedSize Int | Undefined deriving (Eq, Read, Show, Generic, NFData, Out)
+
+
+instance Ord RegionSize where
+  (<=) (BoundedSize sz1) (BoundedSize sz2) = sz1 <= sz2
+  (<=) Undefined _         = error "Invalid comparison"
+  (<=) _         Undefined = error "Invalid comparison"
```

Comment:
I didn't see this being used anywhere. But why not

```
Bounded a <= Bounded b = a <= b
_         <= Undefined = True
```

?

Reply:
I am using undefined as something that can be either finite or infinite, just that we aren't sure about it, so technically it might be wrong to have an ordering against it.

Reply:
Yep, I understood the semantics of `Undefined`, but I don't see why we can't have an ordering over it. Anyway, we can talk about it over Slack later. This and the other `Num` instance jumped out specifically because a runtime error in a type class instance doesn't sound like a good idea...

Reply:
I have changed that to a Semigroup and a Monoid Instance.